packages/transfer/src/index.vue

```diff
@@ -0,0 +1,235 @@
+    >
+      <template #default>
+        <slot name="left-footer"></slot>
+      </template>
+    </transfer-panel>
+    <div class="el-transfer__buttons">
+      <el-button
+        type="primary"
+        :class="['el-transfer__button', hasButtonTexts ? 'is-with-texts' : '']"
+        :disabled="rightChecked.length === 0"
+        @click="addToLeft"
+      >
+        <i class="el-icon-arrow-left"></i>
+        <span v-if="buttonTexts[0] !== undefined">{{ buttonTexts[0] }}</span>
+      </el-button>
+      <el-button
+        type="primary"
+        :class="['el-transfer__button', hasButtonTexts ? 'is-with-texts' : '']"
+        :disabled="leftChecked.length === 0"
+        @click="addToRight"
+      >
+        <span v-if="buttonTexts[1] !== undefined">{{ buttonTexts[1] }}</span>
+        <i class="el-icon-arrow-right"></i>
+      </el-button>
+    </div>
+    <transfer-panel
+      ref="rightPanel"
+      :data="targetData"
+      :render-content="renderContent"
+      :placeholder="panelFilterPlaceholder"
+      :filterable="filterable"
+      :format="format"
+      :filter-method="filterMethod"
+      :title="rightPanelTitle"
+      :default-checked="rightDefaultChecked"
+      :props="props"
+      @checked-change="onTargetCheckedChange"
+    >
+      <template #default>
+        <slot name="right-footer"></slot>
+      </template>
+    </transfer-panel>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType, provide, reactive, ref, toRefs, VNode, watch } from 'vue'
+import { t } from '@element-plus/locale'
+import ElButton from '@element-plus/button/src/button.vue'
+import TransferPanel from './transfer-panel.vue'
+import { useComputedData } from './useComputedData'
+import { useCheckedChange } from './useCheckedChange'
+import { useMove } from './useMove'
+import { Key } from './transfer'
+
+import { UPDATE_MODEL_EVENT } from '@element-plus/utils/constants'
+
+export const CHANGE_EVENT = 'change'
+export const LEFT_CHECK_CHANGE_EVENT = 'left-check-change'
+export const RIGHT_CHECK_CHANGE_EVENT = 'right-check-change'
+
+export default defineComponent({
+  name: 'ElTransfer',
+
+  components: {
+    TransferPanel,
+    ElButton,
+  },
+
+  props: {
+    data: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    titles: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    buttonTexts: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    filterPlaceholder: {
+      type: String,
+      default: '',
+    },
+    filterMethod: {
+      type: Function as PropType<(query: string, item: Record<string, any>) => boolean>,
+    },
+    leftDefaultChecked: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    rightDefaultChecked: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    renderContent: {
+      type: Function as PropType<(option) => VNode>,
```

Comment:
My bad, renderContent is a public API and its first argument is `h` render function. We cannot remove it here for backward compatibility....